dev/AppNotifications/AppNotificationTelemetry.h

```diff
@@ -0,0 +1,319 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <WindowsAppRuntimeInsights.h>
+#include <wrl\wrappers\corewrappers.h>
+
+DECLARE_TRACELOGGING_CLASS(AppNotificationTelemetryProvider,
+    "Microsoft.WindowsAppSDK.Notifications.AppNotificationTelemetry",
+    // {7c1b07ef-a7c0-56d1-5456-385ebd4412b2}
+    (0x7c1b07ef, 0xa7c0, 0x56d1, 0x54, 0x56, 0x38, 0x5e, 0xbd, 0x44, 0x12, 0xb2)); //ELx need a new guid here
+
+
+class AppNotificationTelemetry : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TELEMETRY_CLASS(AppNotificationTelemetry, AppNotificationTelemetryProvider);
+
+public:
+    DEFINE_EVENT_METHOD(RegisterByAPI)(
```

Comment:
What does ByAPI suffix mean in all event names?

Reply:
I replicated the same pattern we used for PushNotification telemetry. We added **byApi** to the events to indicate that these were discreet API calls made by an app.

Reply:
If all events are marked as ByApi, then it isn't needed at all, right?

Reply:
I don't disagree, but we might add more events in the future when we do the comprehensive version and the push notifications event also all have byApi and may also get new events soon. I would think that whatever reasoning was applied when we decided to add ByApI for all push notification events would apply for this situation too.